graph/src/data_source/offchain.rs

```diff
@@ -112,14 +114,14 @@ impl DataSource {
     /// used as the value to be returned to mappings from the `dataSource.address()` host function.
     pub fn address(&self) -> Option<Vec<u8>> {
         match self.source {
-            Source::Ipfs(cid) => Some(cid.to_bytes()),
+            Source::Ipfs(CidFile { cid, path: _ }) => Some(cid.to_bytes()),
```

Comment:
What happens is that we were using `dataSource.address()` to return the CID in byte form. But now the input can be a CID + path to traverse directories. And paths are strings. A couple options I can see:
1. The one taken here, return just the CID and lose the path.
2. Change this to return an utf8 string, which is more symmetrical with the data source creation which takes a string. The host fn is called `address` and returns an `Address` on the AS side which is strange, but we can improve that on graph-ts.

I lean towards 2, wdyt @azf20.

Reply:
we could also just return the entire path {cid}/{path} in bytes?

Reply:
Yes I think I wasn't clear, 2 is returning the entire path as a string encoded as a utf8 byte array.

Reply:
sorry I missed this - yes I think 2 makes sense here.
I had some trouble turning `dataSource.address()` into an ipfsHash I recognised during my testing, but I expect that was user error

Reply:
That's the consensus then, @mangas lets do 2 and this will be ready to merge.